src/python/pants/backend/tools/semgrep/subsystem.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Iterable
+
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.target_types import ConsoleScript
+from pants.engine.rules import Rule, collect_rules
+from pants.engine.target import Dependencies, FieldSet, SingleSourceField, Target
+from pants.engine.unions import UnionRule
+from pants.option.option_types import ArgsListOption, BoolOption, SkipOption
+from pants.util.strutil import softwrap
+
+
+@dataclass(frozen=True)
+class SemgrepFieldSet(FieldSet):
+    required_fields = (SingleSourceField, Dependencies)
+    source: SingleSourceField
+    dependencies: Dependencies
+
+    @classmethod
+    def opt_out(cls, tgt: Target) -> bool:
+        # FIXME: global skip_semgrep field?
+        return False
+
+
+class SemgrepSubsystem(PythonToolBase):
+    name = "Semgrep"
+    options_scope = "semgrep"
+    help = softwrap(
+        """\
```

Comment:
no need to escape the newline for softwrap'ed long strings.

Reply:
(Accidentally pushed to the wrong branch. Fixed in https://github.com/pantsbuild/pants/pull/19685)